### PR TITLE
Improve Naomi/AW input system. Fix custom texture loading race conditions

### PR DESCRIPTION
**Pull request by flyinghead**
- Improve Naomi/AW input system (use game input descriptors)

- Fix race conditions when loading custom textures

- Clear color buffer before drawing imgui background

- Add horizontal screen stretching option

### CODE REVIEW

**Comment by skmp** [comment]
This pull request **fixes 4 alerts** when merging 4b7e4f467795ea67e38e2538c86112e92df18b34 into 7e11e7aff6d704de4ad8ad7531f597df058099ac - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-e65483170f038920bf99272a5f39a76b080752cb)

**fixed alerts:**

* 4 for FIXME comment

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by skmp** [comment]
This pull request **fixes 4 alerts** when merging 408d16b2993302b1c5384b4876cf7e14df328d01 into 7e11e7aff6d704de4ad8ad7531f597df058099ac - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-bac0313965a3798a044f7656399cdf55962509c8)

**fixed alerts:**

* 4 for FIXME comment

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by skmp** [comment]
This pull request **fixes 4 alerts** when merging 47bb509f02802cf6c8a233d249573ba6c118b1a2 into 7e11e7aff6d704de4ad8ad7531f597df058099ac - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-3052da09d314552dcebb1677ce7c628c70cc521f)

**fixed alerts:**

* 4 for FIXME comment

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by skmp** [comment]
This pull request **fixes 4 alerts** when merging d78fc620551dbaa17897e641d993e20bccd1c83b into 7e11e7aff6d704de4ad8ad7531f597df058099ac - [view on LGTM.com](https://lgtm.com/projects/g/reicast/reicast-emulator/rev/pr-9bf7c7ec125234119d5524179eb572e53b72823a)

**fixed alerts:**

* 4 for FIXME comment

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: core/hw/maple/maple_devs.cpp [hunk 1]
```diff
@@ -215,7 +216,12 @@ struct maple_sega_controller: maple_base
 
 			//struct data
 			//3*4
-			w32( 0xfe060f00);
+#if DC_PLATFORM != DC_PLATFORM_ATOMISWAVE
```
Comment (skmp) [inline core/hw/maple/maple_devs.cpp]: We prob. want to subclass here instead of using #defines

File: core/rend/gl4/gles.cpp [hunk 2]
```diff
@@ -662,14 +662,15 @@ static bool RenderFrame()
 		Handle Dc to screen scaling
 	*/
 	float screen_scaling = is_rtt ? 1.f : settings.rend.ScreenScaling / 100.f;
+	float screen_stretching = settings.rend.ScreenStretching / 100.f;
```
Comment (skmp) [inline core/rend/gl4/gles.cpp]: I'd add this as a more generic screen scaling argument - we can do various aspect ratios, considering the NTSC safe area and such, not only anamorphic widescreen

File: core/hw/maple/maple_devs.cpp (hunk 1 above)
Comment (flyinghead) [reply core/hw/maple/maple_devs.cpp]: agreed

File: core/rend/gl4/gles.cpp (hunk 2 above)
Comment (flyinghead) [reply core/rend/gl4/gles.cpp]: yes, we could add a vertical stretching option as well, positive or negative

Comment (skmp) [reply core/rend/gl4/gles.cpp]: See https://github.com/emudev-org/nulldce/blob/39c05500629116f0eb6bcb89a94027e1b987170c/plugs/drkPvr/softRend.cpp#L345 for good values

**Pull request merged**